doc/guides/debugging.md

```diff
@@ -0,0 +1,71 @@
+# Debugging
+
+<!-- type=misc -->
+
+Flag | Impact
+-----|-------
+--debug | Enable debugger agent\nListen on default port (5858)
+--debug=n | Enable debugger agent\nListen on port n.
+--debug-brk | Enable debugger agent\nListen on default port (5858)\nBreak before user code starts.
+--debug-brk=n | Enable debugger agent\nListen on port n\nBreak before user code starts.
+--inspect | Enable inspector agent\nListen on default port (9229).
+--inspect=n | Enable inspector agent\nListen on port n.
+--inspect-brk | Enable inspector agent\nListen on default port (9229)\nBreak before user code starts.
+--inspect-brk=n | Enable inspector agent\nListen on port n\nBreak before user code starts.
+--debug-port=n | Set host and port for the previous. Must come at end.
+`node debug` | Spawn child process to run user script under --debug mode, and use main process to run CLI debugger.
+
+Default ports are 9229 for Inspector, 5858 for Debugger. Default host is 127.0.0.1.
+
+
+## Enable Debugging
+
+When started with the `--debug` or `--debug-brk` switches, Node.js listens
+for debugging commands defined by the [V8 Debugging Protocol][] on a TCP port,
+by default `5858`. Any debugger client which speaks this protocol can connect
+to and debug the running process.
+
+[V8 Debugging Protocol]: https://github.com/v8/v8/wiki/Debugging-Protocol
+
+A running Node.js process can also be instructed to start listening for
+debugging messages by signaling it with `SIGUSR1` (on Linux and OS X).
+
+_Experimental_
+
+When started with the `--inspect` or `--inspect-brk` switches, Node.js listens for
+commands defined by the [Chrome Debugging Protocol][] via websockets, by
+default at `ws://127.0.0.1:9229/node`. Any debugger client which speaks this protocol can
+connect to and debug the running process.
+
+[Chrome Debugging Protocol]: https://developer.chrome.com/devtools/docs/debugger-protocol
+
+## Builtin Debugger Client
+
+Node.js includes a simple debugger client capable of step operations and
+inspection of state. See [../api/debugger.md](../api/debugger.md) for full
+specification of the debugger's capabilities.
+
+When started with the `node debug` command, Node.js runs the specified user script in a
+child process with the `--debug-brk` flag and starts a debugger client which connects to that process.
```

Comment:
Fixed the long lines outside of tables and long URLs. Will switch to an embedded HTML table for better control. Not sure what to do about the long URLs...